File: tests/fixtures/python/adapter_matrix.py
# @generated
from dataclasses import dataclass


@generated
@dataclass
class Report:
    value: int


async def compute(π: int, values: list[int]) -> int:
    # line comment
    total = π * 2
    for value in values:
        if value > 0:
            total += value

    match total:
        case 0:
            return 0
        case _:
            print total

    exec "legacy = True"
    return await helpers.normalize(total)

File: tests/fixtures/rust/adapter_matrix.rs
#![allow(dead_code)]

use std::ptr;

#[generated]
struct Γ {
    value: i32,
}

macro_rules! double {
    ($value:expr) => {
        $value * 2
    };
}

#[automatically_derived]
impl Γ {
    fn compute(&self, π: i32) -> i32 {
        // line comment
        let values = vec![1, 2];
        /* block comment */
        unsafe { ptr::read(&π) }
            + if self.value == 0 {
                double!(π)
            } else {
                values.len() as i32
            }
    }
}
